lib/main.js

```diff
@@ -327,6 +371,22 @@ function signText(text) {
       return ERROR_INTERNAL;
     }
 
+    slotList = PK11_GetAllSlotsForCert(cert, null);
+    if (slotList.isNull()) {
+      log("PK11_GetAllSlotsForCert failed");
+      cleanupSignTextResources(cert, contentInfo, encodedItem, slotList);
+      return ERROR_INTERNAL;
+    }
+    let it = slotList.contents.head;
+    while (it && !it.isNull()) {
+      if (PK11_CheckUserPassword(it.contents.slot, certName.password) != SECSuccess) {
```

Comment:
What happens if the certificate exists on multiple slots? I'm assuming PK11_CheckUserPassword will fail on the slots other than the one the password is for. I think we need to do a couple more things here:
a) only call PK11_CheckUserPassword if (PK11_NeedLogin(slot) && !PK11_IsLoggedIn(slot, nullptr))
b) if the call fails and PR_GetError() == SEC_ERROR_BAD_PASSWORD, don't immediately report that as an error - just keep track that it happened, and if PK11_CheckUserPassword later succeeds, assume the error was for an unrelated slot (if there was never a success, then we should report the authentication failure)